lib/wanda/execution.ex

```diff
@@ -17,15 +17,21 @@ defmodule Wanda.Execution do
       |> Enum.uniq()
       |> Enum.map(&Catalog.get_check/1)
 
-    DynamicSupervisor.start_child(
-      Supervisor,
-      {Server,
-       execution_id: execution_id,
-       group_id: group_id,
-       targets: targets,
-       checks: checks,
-       config: config}
-    )
+    case DynamicSupervisor.start_child(
+           Supervisor,
+           {Server,
+            execution_id: execution_id,
+            group_id: group_id,
+            targets: targets,
+            checks: checks,
+            config: config}
+         ) do
+      {:ok, _} ->
+        :ok
+
+      {:error, _} = error ->
+        error
```

Comment:
Why weren't we happy with the previous one?

Reply:
execution behaviour is defined to return :ok or error, returning a pid is an implementation detail of this specific implementation. we are just interested to know that the execution was started properly

Reply:
Glad to see that I had the same solution mind XD